File: workspace/dump_memory.py
import lancedb
try:
    db = lancedb.connect("/Users/mac/.openclaw/workspace/lancedb_data")
    tbl = db.open_table("structured_memory")
    # Fetch using arrow directly
    ds = tbl.to_arrow()
    records = ds.to_pylist()
    print(f"========= TOTAL RECORDS: {len(records)} =========")
    for i, row in enumerate(records):
        print(f"--- Record {i+1} ---")
        print(f"[{row.get('created_at', '')}]")
        print(row.get('text', 'No Text'))
except Exception as e:
    import traceback
    print(f"Error: {e}")
    traceback.print_exc()

File: workspace/dump_memory.js
const lancedb = require('vectordb'); // or whatever they use
async function run() {
    try {
        const db = await lancedb.connect('/Users/mac/.openclaw/workspace/lancedb_data');
        const tbl = await db.openTable('structured_memory');
        const results = await tbl.query().limit(1000).execute();
        console.log(`========= TOTAL RECORDS: ${results.length} =========`);
        results.forEach((r, i) => {
            console.log(`--- Record ${i+1} ---`);
            console.log(r.text);
        });
    } catch(e) {
        console.error(e);
    }
}
run();
